### DISCUSSION TITLE
Authentication Failed on GitHub Desktop - macOS

### DISCUSSION THREAD

**Discussion by dirkNlerxst** [community/community#67965] [New to GitHub]
Hello! I'm having some issues pushing to my fork of a project. When I go to push upstream, I get an authentication error. I've tried logging in and out, and completely removing Github Desktop and associated folders in Cache/Application Support etc. The repo is not archived, I don't (think?) I use SSH auth. The last portion of the error says to create a PAT so I created one, but I don't know where to put it in GitHub desktop? When I use the sign in under Settings, it opens my browser and I have to log in with my password, my PAT doesn't work. I have a portion of the log below. I am on the latest ARM version of GH Desktop, macOS Ventura 13.5.1, M1 Mac Mini
[github desktop auth error.txt](https://github.com/community/community/files/12702242/github.desktop.auth.error.txt)

Thanks in advance!
![Screenshot 2023-09-22 at 7 45 57 AM](https://github.com/community/community/assets/132940189/a35162c6-a619-496e-98ff-31dc2597eca5)

**Comment by ThidarPhyo** [comment]
up!!! want to know too

**Comment by SumanthSShet** [comment]
To troubleshoot the GitHub Desktop authentication error you are receiving, you can try the following:

Check your PAT permissions. Make sure that the PAT you created has the necessary permissions to push to the repository. You can check the permissions by going to Settings > Developer settings > Personal access tokens.
Re-authenticate with GitHub Desktop. Go to Settings > Accounts and click the Sign out button next to your account. Then, click the Sign in button and sign in using your PAT.
Delete the GitHub Desktop cache. Go to Help > Show Logs in Finder. Open the github.desktop.log file and scroll to the bottom. Find the line that says Cache directory:. This is the path to the GitHub Desktop cache. Close GitHub Desktop and delete the cache directory.
Restart GitHub Desktop.
If you are still having trouble, you can try the following:

Use SSH authentication. SSH authentication is a more secure way to authenticate with GitHub. To use SSH authentication, you will need to generate an SSH key and add it to your GitHub account. You can find instructions on how to generate and add an SSH key to your GitHub account on the GitHub website.
Contact GitHub support. If you have tried all of the above and you are still having trouble, you can contact GitHub support for assistance.
Here are some additional tips for using PATs in GitHub Desktop:

When you sign in to GitHub Desktop using a PAT, you will be prompted to enter your PAT in the password field.
You can also add your PAT to the GitHub Desktop configuration file. To do this, open the GitHub Desktop configuration file in a text editor. The configuration file is located at ~/.config/github-desktop/config. Add the following line to the configuration file:
oauth2_token: YOUR_PAT
Replace YOUR_PAT with your PAT. Then, save and close the configuration file.

Restart GitHub Desktop and try pushing your changes to the repository again.

I hope this helps!

**Comment by steveward** [comment]
@dirkNlerxst can you confirm you have push access to the upstream repository? You mentioned pushing to your fork, but if you're trying to push to the parent I would verify that you have the correct read/write permissions to push to that repository.
